warden/group/manager_sql.go

```diff
@@ -57,16 +57,22 @@ func (m *SQLManager) CreateSchemas() (int, error) {
 	return n, nil
 }
 
+func (m *SQLManager) createGroup(group string) func(tx *sqlx.Tx) error {
+	return func(tx *sqlx.Tx) error {
+		if _, err := tx.Exec(m.DB.Rebind("INSERT INTO hydra_warden_group (id) VALUES (?)"), group); err != nil {
+			return err
```

Comment:
Normal errors do not have stack traces in go, to add a stack trace do `return errors.WithStack(err)`

Reply:
Oh now I got it, I thought that `errors.WithStack(err)` just has to be called once to get the whole stacktrace.

Reply:
Exactly, and it also needs to be called at the location which is the deepest in the stack trace and there only once - otherwise you'll overwrite the stack trace. This is a bit tricky unfortunately. The rule is that we wrap all errors coming from other libraries but don't wrap errors coming from our own. That way we ensure that traces are always as deep as possible in the stack.

Reply:
So [this](https://github.com/ory/hydra/pull/768/files#diff-70ba1b1b4893bb9e2acf833a21da73b7R182) should be actually without a stacktrace?